awx/main/models/unified_jobs.py

```diff
@@ -1512,3 +1512,35 @@ def log_lifecycle(self, state, blocked_by=None):
         else:
             msg = f"{self._meta.model_name}-{self.id} {state.replace('_', ' ')}"
         logger_job_lifecycle.debug(msg, extra=extra)
+
+    @property
+    def launched_by(self):
+        ancestor_job = self.ancestor_job
+        if ancestor_job.launch_type == "manual" or ancestor_job.launch_type == "relaunch":
+            return ancestor_job.created_by
+        elif ancestor_job.launch_type == "scheduled":
+            return ancestor_job.schedule
+        elif ancestor_job.launch_type == "workflow":
+            return ancestor_job.get_workflow_job()
+        elif ancestor_job.launch_type == "dependency":
+            if ancestor_job.job_type_name == "project_update":
+                return ancestor_job.project
+            elif ancestor_job.job_type_name == "inventory_update":
+                return ancestor_job.inventory
```

Comment:
@jbradberry not sure I fully understand why its getting set to the inventory, when really it is being launched as a dependency of the job template launch.

Reply:
I guess these objects are the *closest* approximation to an object we have?

But I'll agree, it's not really "the inventory" that launched it, it's Tower's task system.

Reply:
"inventory" is not really the object that launched it, but i took a close approximation. From what i understand , the task manager was launching the job. But i was not able to figure out what should be returned in this case. ( I might be wrong here )